webapp/scripts/query.js

```diff
@@ -21,16 +24,27 @@ class ModuleTwo {
       return messages["7"]
     }
 
-    initMapset('module_2')
+    initMapset(this.mapset)
 
     // set selection as the query area
-    grass('module_2', `g.copy vector=selection@PERMANENT,selection --overwrite`)
+    grass(this.mapset, `g.copy vector=selection@PERMANENT,selection --overwrite`)
     this.queryArea = 'selection'
 
+    const allVector = listUserVector()
+    // check and add all layers from layer switcher
+    getAllFile(CONTAINER_GEOSERVER, [], 'gpkg').forEach(mapFile => {
+      if (allVector.indexOf(mapFile.slice(mapFile.lastIndexOf('/') + 1, mapFile.lastIndexOf('.'))) < 0) {
+        try {
+          addVector('PERMANENT', mapFile, mapFile.slice(mapFile.lastIndexOf('/') + 1, mapFile.lastIndexOf('.')))
```

Comment:
store long things like `mapFile.slice(mapFile.lastIndexOf('/') + 1, mapFile.lastIndexOf('.'))` in a variable if you reuse them later